src/mono/mono/metadata/appdomain.c

```diff
@@ -1239,13 +1243,16 @@ void
 mono_runtime_register_appctx_properties (int nprops, const char **keys,  const char **values)
 {
 	n_appctx_props = nprops;
-	appctx_keys = g_new0 (gunichar2*, nprops);
-	appctx_values = g_new0 (gunichar2*, nprops);
+	appctx_keys = (char**)keys;
+	appctx_values = (char**)values;
```

Comment:
this won't work.  The values here are passed from `monovm_initialize`, so we can't depend on them being valid after `monovm_initialize` returns - the caller might have deallocated the memory after calling `mono_vm_initialize`.

Reply:
You are right! Fixed.